src/main/host/syscall/handler/mman.rs

```diff
@@ -95,13 +95,22 @@ impl SyscallHandler {
         let addr: usize = addr.try_into().unwrap();
         let addr = ForeignPtr::<()>::from(addr).cast::<u8>();
 
-        // check for truncated flag bits (use u32 instead of i32 to prevent sign extension when
-        // casting from signed to unsigned)
-        if prot as u32 as u64 != prot {
-            warn_once_then_trace!("Ignoring truncated prot flags from mprotect: {prot}");
-        }
-
-        let prot = prot as i32;
+        let Some(prot) = ProtFlags::from_bits(prot) else {
+            for flag in ProtFlags::from_bits_retain(prot)
+                .difference(ProtFlags::all())
+                .iter()
+            {
+                log_once_per_value_at_level!(
+                    flag,
+                    ProtFlags,
+                    log::Level::Warn,
+                    log::Level::Debug,
+                    "Unrecognized prot flag: 0x{:x}",
```

Comment:
I think this can be "Unrecognized prot flag: {:#x}" instead?

Reply:
done